### ISSUE TITLE
Solaar Freezing on Arch Linux w. Gnome 45, when Logi Bolt is connected.

### ISSUE THREAD

**Issue by VeldoraTheDragon** [pwr-Solaar/Solaar#2162]
**Information**

- Solaar version (`solaar --version` or `git describe --tags` if cloned from this repository): solaar 1.1.10
- Distribution: Arch Linux, Gnome 45
- Kernel version (ex. `uname -srmo`): `Linux 6.5.7-arch1-1 x86_64 GNU/Linux`
- Output of `solaar show`:

<details>

```
solaar version 1.1.10

Bolt Receiver
  Device path  : /dev/hidraw6
  USB id       : 046d:C548
  Serial       : 37373735453939354531433434343033
  Has 2 paired device(s) out of a maximum of 6.
  Notifications: wireless, software present (0x000900)
  Device activity counters: 2=134

  1: MX Keys B
     Device path  : None
     WPID         : B363
     Codename     : MX Keys B
     Kind         : keyboard
     Protocol     : unknown (device is offline)
     Serial number: C5A15BE7
     The power switch is located on the (unknown).
     Battery: unknown (device is offline).

  2: Signature M650 L
     Device path  : None
     WPID         : B02A
     Codename     : Logi M650 L
     Kind         : mouse
     Protocol     : HID++ 4.5
     Serial number: 18DEC84B
     Model ID:      B02A00000000
     Unit ID:       18DEC84B
        Bootloader: BL1 38.02.B0013
          Firmware: RBM 16.02.B0013
             Other: 
     The power switch is located on the (unknown).
     Supports 30 HID++ 2.0 features:
         0: ROOT                   {0000} V0     
         1: FEATURE SET            {0001} V0     
         2: DEVICE FW VERSION      {0003} V4     
            Firmware: Bootloader BL1 38.02.B0013 B02AB0706FCD
            Firmware: Firmware RBM 16.02.B0013 B02AB0706FCD
            Firmware: Other   
            Unit ID: 18DEC84B  Model ID: B02A00000000  Transport IDs: {'btleid': 'B02A'}
         3: DEVICE NAME            {0005} V0     
            Name: Signature M650 L
            Kind: mouse
         4: WIRELESS DEVICE STATUS {1D4B} V0     
         5: CONFIG CHANGE          {0020} V0     
         6: DEVICE FRIENDLY NAME   {0007} V0     
            Friendly Name: Logi M650 L
         7: UNIFIED BATTERY        {1004} V3     
            Battery: 95%, discharging.
         8: REPROG CONTROLS V4     {1B04} V5     
            Key/Button Actions (saved): {Middle Button:Mouse Middle Button, Back Button:Mouse Back Button, Forward Button:Mouse Forward Button}
            Key/Button Actions        : {Middle Button:Mouse Middle Button, Back Button:Mouse Back Button, Forward Button:Mouse Forward Button}
            Key/Button Diversion (saved): {Middle Button:Regular, Back Button:Regular, Forward Button:Regular}
            Key/Button Diversion        : {Middle Button:Regular, Back Button:Regular, Forward Button:Regular}
         9: HOSTS INFO             {1815} V2     
            Host 0 (paired): managed-arch0069
        10: XY STATS               {2250} V1     
        11: LOWRES WHEEL           {2130} V0     
            Wheel Reports: HID
            Scroll Wheel Diversion (saved): False
            Scroll Wheel Diversion        : False
        12: ADJUSTABLE DPI         {2201} V2     
            Sensitivity (DPI) (saved): 1000
            Sensitivity (DPI)        : 1000
        13: DFUCONTROL             {00C3} V0     
        14: DEVICE RESET           {1802} V0    internal, hidden, unknown:000010 
        15: unknown:1803           {1803} V0    internal, hidden, unknown:000010 
        16: CONFIG DEVICE PROPS    {1806} V8    internal, hidden, unknown:000010 
        17: unknown:1816           {1816} V0    internal, hidden, unknown:000010 
        18: OOBSTATE               {1805} V0    internal, hidden 
        19: unknown:1830           {1830} V0    internal, hidden, unknown:000010 
        20: unknown:1891           {1891} V7    internal, hidden, unknown:000008 
        21: unknown:18A1           {18A1} V0    internal, hidden, unknown:000010 
        22: unknown:1E00           {1E00} V0    hidden 
        23: unknown:1E02           {1E02} V0    internal, hidden 
        24: unknown:1E22           {1E22} V1    internal, hidden, unknown:000010 
        25: unknown:1602           {1602} V0     
        26: unknown:1EB0           {1EB0} V0    internal, hidden, unknown:000010 
        27: unknown:1861           {1861} V1    internal, hidden, unknown:000010 
        28: unknown:18B1           {18B1} V0    internal, hidden, unknown:000010 
        29: unknown:920A           {920A} V0    internal, hidden, unknown:000010 
     Has 6 reprogrammable keys:
         0: Left Button               , default: Left Click                  => Left Click                
             mse, analytics key events, pos:0, group:1, group mask:empty
             reporting: default
         1: Right Button              , default: Right Click                 => Right Click               
             mse, analytics key events, pos:0, group:1, group mask:empty
             reporting: default
         2: Middle Button             , default: Mouse Middle Button         => Mouse Middle Button       
             mse, reprogrammable, divertable, raw XY, analytics key events, pos:0, group:3, group mask:g1,g2,g3
             reporting: default
         3: Back Button               , default: Mouse Back Button           => Mouse Back Button         
             mse, reprogrammable, divertable, raw XY, analytics key events, unknown:000800, pos:0, group:2, group mask:g1,g2
             reporting: default
         4: Forw
```
</details>

- Contents of `~/.config/solaar/config.yaml` (or `~/.config/solaar/config.json` if `~/.config/solaar/config.yaml` not present):

<details>

```
- 1.1.10
- _NAME: Signature M650 L
  _absent: [hi-res-scroll, hires-smooth-invert, hires-smooth-resolution, hires-scroll-mode, scroll-ratchet, smart-shift, thumb-scroll-invert, thumb-scroll-mode,
    onboard_profiles, report_rate, pointer_speed, speed-change, backlight, backlight-timed, fn-swap, persistent-remappable-keys, disable-keyboard-keys,
    crown-smooth, divert-crown, divert-gkeys, m-key-leds, mr-key-led, multiplatform, change-host, gesture2-gestures, gesture2-divert, gesture2-params, sidetone,
    equalizer, adc_power_management]
  _battery: 4100
  _modelId: B02A00000000
  _serial: 18DEC84B
  _unitId: 18DEC84B
  _wpid: B02A
  divert-keys: {82: 0, 83: 0, 86: 0}
  dpi: 1000
  lowres-scroll-mode: false
  reprogrammable-keys: {82: 82, 83: 83, 86: 86}
- {_NAME: Signature M650 L, _modelId: B02A00000000, _serial: 18DEC84B, _unitId: 18DEC84B, _wpid: B02A}
- {_NAME: MX Keys B, _serial: C5A15BE7, _wpid: B363}
```
</details>

- Errors or warrnings from Solaar:

<details>

```
2023-10-18 14:12:38,578,578     INFO [MainThread] root: version 1.1.10, language en_US (UTF-8)
2023-10-18 14:12:38,669,669     INFO [MainThread] logitech_receiver.diversion: GDK Keymap set up
2023-10-18 14:12:38,669,669  WARNING [MainThread] logitech_receiver.diversion: rules cannot access modifier keys in Wayland, accessing process only works on GNOME with Solaar Gnome extension installed
2023-10-18 14:12:38,669,669  WARNING [MainThread] logitech_receiver.diversion: cannot create uinput device: "/dev/uinput" cannot be opened for writing
2023-10-18 14:12:38,678,678     INFO [MainThread] solaar.upower: connected to system dbus, watching for suspend/resume events
2023-10-18 14:12:38,693,693     INFO [MainThread] solaar.ui.notify: starting desktop notifications
2023-10-18 14:12:38,696,696  WARNING [MainThread] solaar.ui.icons: failed to detect icons
2023-10-18 14:12:38,716,716     INFO [MainThread] solaar.listener: starting receiver listening threads
2023-10-18 14:12:38,795,795  WARNING [MainThread] hidapi.udev: Report Descriptor not processed for BID 0003 VID 00001050 PID 00000402: Expecting 64 usages but got 1
2023-10-18 14:12:38,799,799     INFO [MainThread] hidapi.udev: Found device BID 0003 VID 0000046D PID 0000C548 HID++ True True USB 2 2
2023-10-18 14:12:38,799,799     INFO [MainThread] solaar.listener: receiver event add DeviceInfo(path='/dev/hidraw6', bus_id=3, vendor_id='046D', product_id='C548', interface=2, driver='hid-generic', manufacturer=None, product=None, serial='', release=None, isDevice=None, hidpp_short=True, hidpp_long=True)
2023-10-18 14:12:38,799,799     INFO [MainThread] hidapi.udev: OPEN PATH /dev/hidraw6
2023-10-18 14:12:38,799,799     INFO [MainThread] logitech_receiver.base: New lock 17
2023-10-18 14:12:38,801,801     INFO [ReceiverListener:hidraw6] logitech_receiver.listener: started with <BoltReceiver(/dev/hidraw6,17)> (17)
2023-10-18 14:12:38,801,801     INFO [ReceiverListener:hidraw6] solaar.listener: <BoltReceiver(/dev/hidraw6,17)>: notifications listener has started (17)
2023-10-18 14:12:38,801,801     INFO [ReceiverListener:hidraw6] logitech_receiver.base: New lock <_ThreadedHandle(/dev/hidraw6)>
2023-10-18 14:12:38,801,801  WARNING [MainThread] hidapi.udev: Report Descriptor not processed for BID 0003 VID 00000BDA PID 00001100: Expecting 192 usages but got 15
2023-10-18 14:12:38,801,801  WARNING [MainThread] hidapi.udev: Report Descriptor not processed for BID 0018 VID 000004F3 PID 0000320C: Expecting 4 usages but got 2
2023-10-18 14:12:38,803,803     INFO [ReceiverListener:hidraw6] logitech_receiver.receiver: <BoltReceiver(/dev/hidraw6,17)>: receiver notifications enabled => ('wireless', 'software present')
2023-10-18 14:12:38,810,810     INFO [ReceiverListener:hidraw6] solaar.listener: status_changed <BoltReceiver(/dev/hidraw6,17)>: present, No paired devices. (0) 
2023-10-18 14:12:39,818,818     INFO [ReceiverListener:hidraw6] logitech_receiver.receiver: <BoltReceiver(/dev/hidraw6,17)>: found new device 1 (B363)
2023-10-18 14:12:46,550,550  WARNING [ReceiverListener:hidraw6] logitech_receiver.base: (17) timeout (6.73/4.00) on device 1 ping
2023-10-18 14:12:56,556,556    ERROR [ReceiverListener:hidraw6] logitech_receiver.base: lock on handle 17 not acquired, probably due to timeout
2023-10-18 14:13:00,563,563  WARNING [ReceiverListener:hidraw6] logitech_receiver.base: (17) timeout (4.00/4.00) on device 1 ping
2023-10-18 14:12:39,818,818     INFO [ReceiverListener:hidraw6] solaar.listener: connection Notification(10,1,41,10,4163B3) for <Device(1,B363,MX Keys B,C5A15BE7)> (keyboard)
2023-10-18 14:12:46,556,556     INFO [MainThread] hidapi.udev: OPEN PATH /dev/hidraw6
2023-10-18 14:13:00,564,564     INFO [ReceiverListener:hidraw6] solaar.listener: status_changed <BoltReceiver(/dev/hidraw6,17)>: present, 1 paired device. (0) 
2023-10-18 14:13:04,576,576  WARNING [ReceiverListener:hidraw6] logitech_receiver.base: (17) timeout (4.00/4.00) on device 1 ping
2023-10-18 14:13:08,584,584  WARNING [ReceiverListener:hidraw6] logitech_receiver.base: (17) timeout (4.01/4.00) on device 1 ping
2023-10-18 14:13:00,566,566     INFO [ReceiverListener:hidraw6] solaar.listener: status_changed <Device(1,B363,MX Keys B,C5A15BE7)>: paired offline, {'LINK ENCRYPTED': True} (0) 
2023-10-18 14:13:08,586,586     INFO [AsyncUI] hidapi.udev: OPEN PATH /dev/hidraw6
2023-10-18 14:13:12,596,596  WARNING [MainThread] logitech_receiver.base: (20) timeout (4.01/4.00) on device 1 ping
2023-10-18 14:13:12,602,602     INFO [ReceiverListener:hidraw6] logitech_receiver.receiver: <BoltReceiver(/dev/hidraw6,17)>: found new device 2 (B02A)
2023-10-18 14:13:22,603,603    ERROR [ReceiverListener:hidraw6] logitech_receiver.base: lock on handle 17 not acquired, probably due to timeout
2023-10-18 14:13:32,709,709    ERROR [ReceiverListener:hidraw6] logitech_receiver.base: lock on handle 17 not acquired, probably due to timeout
Killed
```
</details>

**Describe the bug**
When having a logitech bolt connected to the device and then opening solaar, it freezes after a few seconds.
If you press "wait", it resumes and then freezes again after a few seconds.

**To Reproduce**
1. Connect a logitech bolt receiver to the device. (unifying receiver works w.o. problems)
2. Open solaar
3. Wait a few seconds for freeze

**Additional context**
In the journal log, there's a message appearing, I don't know if this is the same problem or a different one:
```
solaar[29742]: gtk_widget_get_scale_factor: assertion 'GTK_IS_WIDGET (widget)' failed
```

**Comment by pfps** [comment]
The error is very likely related to the offline device.

Please try with `solaar -ddd`.  Also please then try with the MX Keys B active.  

Thanks.

**Comment by pfps** [comment]
So when the keyboard is active there is no problem at all.  This strongly indicates that the keyboard is providing a ping response that Solaar does not understand.   I'll need the output of `solaar -ddd` when the keyboard is not active to debug the problem.

**Comment by pfps** [comment]
OK, Solaar is definitely receiving an error return from the device that it does not expect, and thus ignores.

**Comment by pfps** [comment]
Fixed in #2149 but this has not yet been added to the distributed version of Solaar so you will have to install Solaar from this repository and run it.
